build/build.py

```diff
@@ -381,10 +381,14 @@ async def main():
   bazel_command_base.append("run")
 
   if args.python_version:
-    logging.debug("Hermetic Python version: %s", args.python_version)
-    bazel_command_base.append(
-        f"--repo_env=HERMETIC_PYTHON_VERSION={args.python_version}"
-    )
+    # Do not add --repo_env=HERMETIC_PYTHON_VERSION with default args.python_version
```

Comment:
Wouldn't a simpler solution be to just (a) add free-threading to the list of allowed `python_version` args and (b) pass it via `--python_version` rather than a bazel option?

Reply:
Yes, I could use `--python_version` option instead of `--bazel_options=--repo_env=HERMETIC_PYTHON_VERSION=...` and maybe to specify other hermetic options with bazel options. However, in general it will still be a problem if user wants to use another version that is not included into python version choices list and specify manually a `HERMETIC_PYTHON_VERSION` with repo_env.

Possible, alternatively, we could set the python_version default value: https://github.com/jax-ml/jax/blob/43588c66047b6a14e38896d05172381d232df43e/build/build.py#L73 as None and set it in the code
after checking `args.bazel_options` ?

Reply:
I think this can be simpler.

Use `python_version` to pass Python versions. That's what it's for. You don't need to also set `HERMETIC_PYTHON_VERSION`.

We probably should also change the flag to allow novel Python versions, rather than only allowing a particular allowlist of versions.

Reply:
@hawkinsp Peter, I made the change you suggested, let me know whether it looks what you thought, thanks!